src/nodes.coffee

```diff
@@ -706,19 +713,21 @@ exports.Call = class Call extends Base
   # splatArgs is an array of CodeFragments to put into the 'apply'.
   compileSplat: (o, splatArgs) ->
     if this instanceof SuperCall
-      return [].concat @makeCode("#{ @superReference o }.apply(#{@superThis(o)}, "),
-        splatArgs, @makeCode(")")
+      reference = @superReference o
+      if reference is 'super' or reference.indexOf('super.') is 0
+        return [].concat @makeCode("#{reference}(..."), splatArgs, @makeCode ')'
+      else
+        return [].concat @makeCode("#{reference}.apply(#{@superThis(o)}, "),
+          splatArgs, @makeCode(")")
 
     if @isNew
-      idt = @tab + TAB
-      return [].concat @makeCode("""
-        (function(func, args, ctor) {
-        #{idt}ctor.prototype = func.prototype;
-        #{idt}var child = new ctor, result = func.apply(child, args);
-        #{idt}return Object(result) === result ? result : child;
-        #{@tab}})("""),
-        (@variable.compileToFragments o, LEVEL_LIST),
-        @makeCode(", "), splatArgs, @makeCode(", function(){})")
+      complex = @variable.isComplex()
```

Comment:
Compiles `new` calls with splats to ES2015 splats. Again, a changing to ES2015 splats generally would supersede this.